src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const STATES = [];
+  let copy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
Use destructuring for `action`
```suggestion
  for (const action of actions) {
   const { type, ........ } = action;
   
    switch (type) {
```